Replicator/ReplicatorOptions.hh

```diff
@@ -315,26 +321,60 @@ namespace litecore { namespace repl {
             back.pullFilter = collOpts.pullFilter;
             back.callbackContext = collOpts.callbackContext;
         }
-        trimDisabledCollections();
     }
 
-    inline void Options::trimDisabledCollections() {
+    inline void Options::sanitize() const {
         for (size_t i = collectionOpts.size(); i-- > 0; ) {
             if (collectionOpts[i].push == kC4Disabled
                 && collectionOpts[i].pull == kC4Disabled) {
-                collectionOpts.erase(collectionOpts.begin() + i);
+                throw error(error::LiteCore, error::InvalidParameter,
+                            "Invalid replicator configuration: a collection with both push and pull disabled");
             }
         }
 
         // Assertion: collectionOpts contains no disabled collections
         // (of which both push and pull are disabled)
 
+        // Do not allow active and passive to be mixed in the same replicator.
+        
+        unsigned passCount = 0;
+        unsigned actiCount = 0;
         for (auto c: collectionOpts) {
-            if (c.push == kC4Passive && c.pull == kC4Passive) {
-                hasPassiveCollection = true;
+            if (c.push == kC4Passive)
+                ++passCount;
+            else if (c.push > kC4Passive)
+                ++actiCount;
+            if (c.pull == kC4Passive)
+                ++passCount;
+            else if (c.pull > kC4Passive)
+                ++actiCount;
+
+            if (passCount * actiCount > 0) {
+                throw error(error::LiteCore, error::InvalidParameter,
+                            "Invalid replicator configuration: the collection list includes"
+                            " both passive and active ReplicatorMode");
             }
-            if (c.push != kC4Passive || c.pull != kC4Passive) {
-                isAllPassive = false;
+        }
+        isActive = actiCount > 0;
+        
+        if (collectionOpts.size() == 1) {
+            auto spec = collectionPathToSpec(collectionOpts[0].collectionPath);
+            if (spec == kC4DefaultCollectionSpec) {
+                collectionAware = false;
```

Comment:
This collectionAware is only valid for active replicator. I don't know now that any handler for passive replicators will need to use this flag or not.

Reply:
From slack: @jianminzhao told me that collectionAware do not used by passive replicator.

Reply:
I think this property is misnamed anyway.  passive will not need to use this flag, but rather it will need to search for the default collection in the list of collections if necessary, which is something I am working on.  What this flag is really indicating is "legacy mode or not"

Reply:
As I weighed it against "legacy mode," I found legacy mode may not stick as we move on. As time goes, we have new legacy things. This is a particular legacy when we don't have the notion of collection, and hence, whenever collection is required in a context, default collection is substituted.
As for passive replicator, we deal with the legacy based on the request. If the request does not specify "collection," default collection is assumed and it is equivalent to contemporary (vs legacy) ['default']. Then, of course, we need to find whether the config is ready for this collection. In my current task, I added a mapping from collectionSpec to index in Options.